lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -24,6 +24,190 @@
   [src getCharacters:(unichar *)&result[0] range:NSMakeRange(0, size)];
   return result;
 }
+std::vector<std::u16string> getAvailableLocalesVector() {
+    NSArray<NSString *> *availableLocales = [NSLocale availableLocaleIdentifiers];
+    std::vector<std::u16string> availableLocalesVector;
+      for (id object in availableLocales) {
+        std::u16string u16StringFromVector = nsStringToU16String(object);
+        availableLocalesVector.push_back(u16StringFromVector);
+      }
+    return availableLocalesVector;
+}
```

Comment:
Now that I think about it, it might be cleaner to put the static list of available locales in this function instead, so that any caller of the function will just get a reference to the pre-computed version. We can initialise it using a lambda as follows:

```suggestion
const std::vector<std::u16string>& getAvailableLocalesVector() {
    static const std::vector<std::u16string> *availableLocalesVector = []{
      NSArray<NSString *> *availableLocales = [NSLocale availableLocaleIdentifiers];
      // Intentionally leaked to avoid destruction order problems.
      auto *vec = new std::vector<std::u16string>();
      for (id str in availableLocales)
        vec.push_back(nsStringToU16String(str));
      return vec;
      }();
    return *availableLocalesVector;
```

The above also eliminates the intermediate variable `u16StringFromVector`, which is more efficient because we get a move instead of a copy of the string. It also means that this function can now be used by multiple other functions, without each of them needing to store their own copy. Finally, it avoids issues with static destruction order by just leaking the vector.